core/src/test/java/org/testcontainers/junit/wait/strategy/HttpWaitStrategyTest.java

```diff
@@ -59,7 +59,9 @@ protected void waitUntilReady() {
                 super.waitUntilReady();
                 ready.set(true);
             }
-        }.forResponsePredicate(s -> s.equals(GOOD_RESPONSE_BODY));
+        }
+        .forResponsePredicate(s -> s.equals(GOOD_RESPONSE_BODY))
```

Comment:
please add a test where both `forStatusCodeMatching` and `withStatusCode` are used (to check that it's not being overwritten)

Reply:
Good call. My code is actually failing! :p 
I've been wondering how to add a test and by your question I actually understood how to write. (Was so easy that I feel dumb not having found that before...)

Reply:
Done

Reply:
But we're still missing a test where we call 
```
forHttp()
    .forStatusCode(200) // <- 
    .forStatusCode(205) // <- 
   .forStatusCodeMatching(it -> it > 400)
```

Reply:
So should I clone the class or just change the `buildWaitStrategy` method?

Reply:
just add a test method which is not based on `buildWaitStrategy`'s result

Reply:
I did it. A bit differently though so I can reuse most of the existing code and that could help in the future.
I hope it's ok for you